### PR TITLE
[RFC] spellfile: follow the XDG spec

### LINKED ISSUE #3535: Spell#LoadFile() error still mentions ~/.nvim
```
spellfile#LoadFile(): There is no writable spell directory.
Shall I create /home/user/.nvim/spell
```

### PR DESCRIPTION
**Pull request by fmoralesc**
Closes #3535

### CODE REVIEW

File: runtime/autoload/spellfile.vim [hunk 1]
```diff
@@ -197,8 +197,12 @@ endfunc
 
 function! spellfile#WritableSpellDir()
   if has("unix")
-    " For Unix always use the $HOME/.nvim directory
-    return $HOME . "/.nvim/spell"
+    " For Unix always use the $XDG_DATA_HOME/nvim/site directory
+    if exists('$XDG_DATA_HOME')
+	return $HOME . $XDG_DATA_HOME . "/nvim/site/spell"
```
Comment (justinmk) [inline runtime/autoload/spellfile.vim]: I didn't think it necessary to prepend HOME to XDG_FOO?

**Comment by justinmk** [comment]
Does the spellfile internal C logic auto-create the directory returned by `spellfile#WritableSpellDir()`?

File: runtime/autoload/spellfile.vim (hunk 1 above)
Comment (fmoralesc) [inline runtime/autoload/spellfile.vim]: :+1: My bad.

**Comment by fmoralesc** [comment]
@justinmk What internal C logic? This plugs into the `SpellFileMissing` event, but otherwise everything is handled by the viml script. The creation of the directory is handled in lines 41-58.

File: runtime/autoload/spellfile.vim [hunk 2]
```diff
@@ -197,8 +197,12 @@ endfunc
 
 function! spellfile#WritableSpellDir()
   if has("unix")
-    " For Unix always use the $HOME/.nvim directory
-    return $HOME . "/.nvim/spell"
+    " For Unix always use the $XDG_DATA_HOME/nvim/site directory
```
Comment (justinmk) [inline runtime/autoload/spellfile.vim]: UNIX check is not needed anymore

Comment (fmoralesc) [inline runtime/autoload/spellfile.vim]: :+1:

File: runtime/autoload/spellfile.vim [hunk 3]
```diff
@@ -196,9 +196,11 @@ function! spellfile#GetDirChoices()
...
 
 function! spellfile#WritableSpellDir()
-  if has("unix")
-    " For Unix always use the $HOME/.nvim directory
-    return $HOME . "/.nvim/spell"
+  " Always use the $XDG_DATA_HOME/nvim/site directory
+  if exists('$XDG_DATA_HOME')
+    return $XDG_DATA_HOME . "/nvim/site/spell"
+  else
+    return $HOME . "/.local/share/nvim/site/spell"
```
Comment (shazow) [inline runtime/autoload/spellfile.vim]: This addition circumvents the remaining body of this function, looks like a bug.

**Comment by shazow** [comment]
I'm pretty sure this PR introduced a bug, see https://github.com/neovim/neovim/pull/3540/files#r49680916.

Not sure of the full consequences of this, but it seems like it always assumes it failed to find the spell directory even if it already exists. Let me know if you'd like for me to open an issue for this.

File: runtime/autoload/spellfile.vim (hunk 3 above)
Comment (justinmk) [inline runtime/autoload/spellfile.vim]: @shazow Thanks! I wonder why the original "unix" logic was there...

Comment (justinmk) [inline runtime/autoload/spellfile.vim]: @shazow If you delete this if-else so that it falls through, do things seem to work?

Comment (ZyX-I) [inline runtime/autoload/spellfile.vim]: @justinmk The remaining body is neither correct. One should not iterate over `split(&rtp, ',')`, there is `globpath()`. If `globpath()` is not to be used, then code must do unescaping and split on rather lengthy regex (you may see examples in VAM and pathogen code, also in many others).

Comment (ZyX-I) [inline runtime/autoload/spellfile.vim]: (Dunno whether it is expected or not, but `globpath(&runtimepath, '')` is doing exactly this.)

Comment (shazow) [inline runtime/autoload/spellfile.vim]: I'm not convinced that this is the cause of my bug (probably fairly unrelated), but this is worth fixing either way. :)

Comment (ZyX-I) [inline runtime/autoload/spellfile.vim]: @shazow I did not say this is the cause of your bug.

Comment (shazow) [inline runtime/autoload/spellfile.vim]: @ZyX-I I was responding to @justinmk.

Comment (ZyX-I) [inline runtime/autoload/spellfile.vim]: @shazow This is why it is good to at least use @mentions at the start. Or quote comment you replied to. One of the reasons I have @mention in most of my replies.

**Pull request merged**